File: surf-cloud-api/src/main/java/dev/slne/surf/cloud/api/netty/exception/SurfNettyRegisterPacketException.java
package dev.slne.surf.cloud.api.netty.exception;

import java.io.Serial;

public class SurfNettyRegisterPacketException extends SurfNettyPacketException {

  @Serial
  private static final long serialVersionUID = -5234200008867680129L;

  public SurfNettyRegisterPacketException(String message) {
    super(message);
  }

  public SurfNettyRegisterPacketException(String message, Throwable cause) {
    super(message, cause);
  }

}

File: surf-cloud-api/surf-cloud-api-common/src/main/kotlin/dev/slne/surf/cloud/api/common/netty/exception/SurfNettyRegisterPacketException.kt
package dev.slne.surf.cloud.api.common.netty.exception

import java.io.Serial

/**
 * Exception indicating a failure to register a Netty packet.
 */
class SurfNettyRegisterPacketException : SurfNettyPacketException {
    constructor(message: String?) : super(message)
    constructor(message: String?, cause: Throwable?) : super(message, cause)

    companion object {
        @Serial
        private val serialVersionUID = -5234200008867680129L
    }
}
